src/ripple/app/misc/NetworkOPs.h

```diff
@@ -123,22 +128,41 @@ class NetworkOPs : public InfoSub::Source
     virtual void
     submitTransaction(std::shared_ptr<STTx const> const&) = 0;
 
-    /**
-     * Process transactions as they arrive from the network or which are
-     * submitted by clients. Process local transactions synchronously
+    /** Process a transaction.
+     *
+     * The transaction has been submitted either from the peer network or
+     * from a client. For client submissions, there are 3 distinct behaviors:
+     *   1) sync (default): process transactions in a batch immediately,
+     *       and return only once the transaction has been processed.
+     *   2) async: Put transaction into the batch for the next processing
+     *       interval and return immediately.
+     *   3) wait: Put transaction into the batch for the next processing
+     *       interval and return only after it is processed.
      *
-     * @param transaction Transaction object
+     * @param transaction Transaction object.
      * @param bUnlimited Whether a privileged client connection submitted it.
-     * @param bLocal Client submission.
-     * @param failType fail_hard setting from transaction submission.
+     * @param sync Client submission synchronous behavior type requested.
+     * @param bLocal Whether submitted by client (local) or peer.
+     * @param failType Whether to fail hard or not.
      */
     virtual void
     processTransaction(
         std::shared_ptr<Transaction>& transaction,
         bool bUnlimited,
+        RPC::SubmitSync sync,
         bool bLocal,
         FailHard failType) = 0;
 
+    /** Apply transactions in batches.
+     *
+     * Only a single batch unless drain is set.
+     *
+     * @param drain Whether to process batches until none remain.
+     * @return Whether any transactions were processed.
+     */
+    virtual bool
+    transactionBatch(bool const drain) = 0;
```

Comment:
*nitpick*:  When passing by value, a `const` qualifier is an implementation detail.  It has use in the definition (this parameter will not change within the definition).  But it is just noise in the declaration.  The caller knows the definition won't change the value the caller sees because it is passed by value.

Recommendation:  Remove `const` only from the declaration, but not the definition.  Here and line 289 of NetworkOPs.cpp.

_Note:_ This is a lone exception to the rule that the prototype of the declaration and definition must be the same.  Name manglers know this rule.  The `const` qualifier on a by-value parameter has no impact on the mangling of the function name, and thus the presence/absence can not be overloaded on.

Reply:
fixed